src/interpreter/Interpreter.ts

```diff
@@ -84,6 +76,13 @@ export class Interpreter {
       this.interpretAndCombineSchema(schema.else, model, schema, interpreterOptions);
 
       interpretNot(schema, model, this, interpreterOptions);
+
+      //All schemas of type model object or enum MUST have ids
+      if (isModelObject(model) === true || isEnum(model) === true) {
```

Comment:
So we end with helper functions to check what type model has 😅 Creating an EnumModel class or the ObjectModel class would be easier in this way as I suggested in long time ago :P

Reply:
And we must remember, `enum` is a generic keyword, it can be also combined with `object` type.

Reply:
It can yea, which also makes the separate classes difficult to work with.

Reply:
But we definitely have to keep an eye out for whether we can reduce the complexity of the model in some way in the future.

Reply:
we can have `modelType` in `CommonModel` and then don't need check it in every place.

Reply:
Hmm, we could yea, not sure it is needed when we have the support functions? 🤔 

But feel free to create an issue 🙂